types/reactjs-localstorage/index.d.ts

```diff
@@ -0,0 +1,14 @@
+// Type definitions for reactjs-localstorage 1.0
+// Project: https://github.com/null-none/react-localstorage
+// Definitions by: Rick Wilson <https://github.com/rwilson504>
+//                 Dan Cox <https://github.com/powerappsdev>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export namespace reactLocalStorage {
+    function set(key: string, value: any): any;
```

Comment:
this is better `set<T = any>(key: string, value: T): T;`

Reply:
localStorage['key'] always returns a string from the browser.

https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage